runtime/gc_base/IndexableObjectAllocationModel.cpp

```diff
@@ -251,6 +259,24 @@ MM_IndexableObjectAllocationModel::layoutDiscontiguousArraylet(MM_EnvironmentBas
 	if (NULL != spine) {
 		switch (_layout) {
 		case GC_ArrayletObjectModel::Discontiguous:
+#if defined(J9VM_GC_ENABLE_DOUBLE_MAP)
+			if (extensions->indexableObjectModel.isDoubleMappingEnabled() && extensions->isVLHGC()) {
+				Assert_MM_true(arrayoidIndex == _numberOfArraylets);
+				UDATA arrayletLeafCount = extensions->indexableObjectModel.numArraylets(spine);
+				UDATA sizeInElements = extensions->indexableObjectModel.getSizeInElements(spine);
+				Assert_MM_true(sizeInElements == 0 || arrayletLeafCount > 0);
+
+				if ((arrayletLeafCount > 1) && (sizeInElements > 0)) {
```

Comment:
some, if not all things are implied since we are in Discontiguous case

Reply:
This check is for the corner case where there's exactly one arraylet leaf, so we wouldn't need to double map. Was that what you meant?